kolena/_experimental/object_detection/dataset.py

```diff
@@ -0,0 +1,246 @@
+from kolena._experimental.object_detection.utils import filter_inferences
+from kolena.annotation import LabeledBoundingBox
+from kolena.annotation import ScoredLabel
+from kolena.annotation import ScoredLabeledBoundingBox
+from kolena.errors import IncorrectUsageError
+from kolena.metrics import InferenceMatches
+from kolena.metrics import match_inferences
+from kolena.workflow.metrics import match_inferences_multiclass
+from kolena.workflow.metrics import MulticlassInferenceMatches
+
+
+def single_class_datapoint_metrics(bbox_matches: InferenceMatches, thresholds: float) -> Dict[str, Any]:
+    tp = [inf for _, inf in bbox_matches.matched if inf.score >= thresholds]
+    fp = [inf for inf in bbox_matches.unmatched_inf if inf.score >= thresholds]
+    fn = bbox_matches.unmatched_gt + [gt for gt, inf in bbox_matches.matched if inf.score < thresholds]
+    scores = [inf.score for inf in tp + fp]
+    return dict(
+        TP=tp,
+        FP=fp,
+        FN=fn,
+        matched_inference=[inf for _, inf in bbox_matches.matched],
+        unmatched_ground_truth=[gt for gt in bbox_matches.unmatched_gt],
+        unmatched_inference=bbox_matches.unmatched_inf,
+        count_TP=len(tp),
+        count_FP=len(fp),
+        count_FN=len(fn),
+        has_TP=len(tp) > 0,
+        has_FP=len(fp) > 0,
+        has_FN=len(fn) > 0,
+        max_confidence_above_t=max(scores) if len(scores) > 0 else None,
+        min_confidence_above_t=min(scores) if len(scores) > 0 else None,
+        thresholds=thresholds,
+    )
+
+
+def multiclass_datapoint_metrics(
+    bbox_matches: MulticlassInferenceMatches,
+    thresholds: Dict[str, float],
+) -> Dict[str, Any]:
+    tp = [inf for _, inf in bbox_matches.matched if inf.score >= thresholds[inf.label]]
+    fp = [inf for inf in bbox_matches.unmatched_inf if inf.score >= thresholds[inf.label]]
+    fn = [gt for gt, _ in bbox_matches.unmatched_gt] + [
+        gt for gt, inf in bbox_matches.matched if inf.score < thresholds[inf.label]
+    ]
+    unmatched_ground_truth = [
+        LabeledBoundingBox(
+            label=gt.label,
+            top_left=gt.top_left,
+            bottom_right=gt.bottom_right,
+            predicted_label=inf.label,  # type: ignore[call-arg]
+            predicted_score=inf.score,  # type: ignore[call-arg]
+        )
+        if inf
+        else gt
+        for gt, inf in bbox_matches.unmatched_gt
+    ]
+    confused = [
+        ScoredLabeledBoundingBox(
+            label=inf.label,
+            score=inf.score,
+            top_left=inf.top_left,
+            bottom_right=inf.bottom_right,
+            actual_label=gt.label,  # type: ignore[call-arg]
+        )
+        for gt, inf in bbox_matches.unmatched_gt
+        if inf is not None and inf.score >= thresholds[inf.label]
+    ]
+    scores = [inf.score for inf in tp + fp]
+    inference_labels = {inf.label for _, inf in bbox_matches.matched} | {
```

Comment:
nit: maybe use `union` rather than `|` for easier reading and consistent style with the other instance